src/editortoolkit.cpp

```diff
@@ -0,0 +1,339 @@
+                assert(pi);
+                pi->AdjustPitchByOffset(-2 * lineDiff); // One line -> 2 pitches
+            } 
+        }
+
+        return true;
+    }
+    return false;
+}
+
+bool EditorToolkit::Insert(std::string elementType, std::string startid, std::string endid)
+{
+    LogMessage("Insert!");
+    if (!m_doc->GetDrawingPage()) return false;
+    Object *start = m_doc->GetDrawingPage()->FindChildByUuid(startid);
+    Object *end = m_doc->GetDrawingPage()->FindChildByUuid(endid);
+    // Check if both start and end elements exist
+    if (!start || !end) {
+        LogMessage("Elements start and end ids '%s' and '%s' could not be found", startid.c_str(), endid.c_str());
+        return false;
+    }
+    // Check if it is a LayerElement
+    if (!dynamic_cast<LayerElement *>(start)) {
+        LogMessage("Element '%s' is not supported as start element", start->GetClassName().c_str());
+        return false;
+    }
+    if (!dynamic_cast<LayerElement *>(end)) {
+        LogMessage("Element '%s' is not supported as end element", start->GetClassName().c_str());
+        return false;
+    }
+
+    Measure *measure = dynamic_cast<Measure *>(start->GetFirstParent(MEASURE));
+    assert(measure);
+    if (elementType == "slur") {
+        Slur *slur = new Slur();
+        slur->SetStartid(startid);
+        slur->SetEndid(endid);
+        measure->AddChild(slur);
+        m_doc->PrepareDrawing();
+        return true;
+    }
+    return false;
+}
+
+bool EditorToolkit::Set(std::string elementId, std::string attrType, std::string attrValue)
+{
+    if (!m_doc->GetDrawingPage()) return false;
+    Object *element = m_doc->GetDrawingPage()->FindChildByUuid(elementId);
+    bool success = false;
+    if (Att::SetAnalytical(element, attrType, attrValue))
+        success = true;
+    else if (Att::SetCmn(element, attrType, attrValue))
+        success = true;
+    else if (Att::SetCmnornaments(element, attrType, attrValue))
+        success = true;
+    else if (Att::SetCritapp(element, attrType, attrValue))
+        success = true;
+    else if (Att::SetExternalsymbols(element, attrType, attrValue))
+        success = true;
+    else if (Att::SetGestural(element, attrType, attrValue))
+        success = true;
+    else if (Att::SetMei(element, attrType, attrValue))
+        success = true;
+    else if (Att::SetMensural(element, attrType, attrValue))
+        success = true;
+    else if (Att::SetMidi(element, attrType, attrValue))
+        success = true;
+    else if (Att::SetPagebased(element, attrType, attrValue))
+        success = true;
+    else if (Att::SetShared(element, attrType, attrValue))
+        success = true;
+    else if (Att::SetVisual(element, attrType, attrValue))
+        success = true;
+    if (success) {
+        m_doc->PrepareDrawing();
+        m_doc->GetDrawingPage()->LayOut(true);
+        return true;
+    }
+    return false;
+}
+
+std::string EditorToolkit::GetNeumeInfo(std::string elementId)
```

Comment:
Something looks weird to me with this here. Why can you use GetElementAttr() and look at `@intm`? There is something unclear with this attribute. What is this purpose? Could you explain this? I can see it generated and then save. This is the first thing to clarify. Normally loading a file in Verovio should not change the content. In any case, I do not think that calculating such values should happen in this class.

Reply:
I'm not sure what you're referring to here. There shouldn't be anything reading `@intm`, let alone saving it when a file is loaded.